### ISSUE TITLE
[Bug report] The header anchor should stay outside of the heading tag

### ISSUE THREAD

**Issue by Vasile-Peste** [vuepress/vuepress-next#1363]
### Description

The header anchors `#` are placed inside the heading tags (h1, h2, h3...).
This behaviour is not actually correct, since the heading tag should explicitly indicate the nature of the content as a title, and `#` shouldn't be part of it.

<img width="597" alt="Screenshot 2023-06-15 at 00 29 02" src="https://github.com/vuepress/vuepress-next/assets/13590581/bd3967a7-4556-4bce-b3d5-9a10ba43b3f4">

Apart from the semantic issue, there is also a more problematic issue: crawlers, some social media crawler may take the title from the h1 heading tag to create a post preview. This means that the preview title of the post will contain the "#" character.

This is an example of a VuePress post preview on LinkedIn

<img width="797" alt="Screenshot 2023-06-15 at 00 34 02" src="https://github.com/vuepress/vuepress-next/assets/13590581/40f04bbf-66fc-48a2-8303-d7a53865114d">

To reproduce I have a live link
https://www.mida.org/posts/reactive-programming-in-financial-markets/

The website above is built with VuePress 1 but the issue persists also on VuePress 2.

Thank you,
Vasile

### Reproduction

https://www.mida.org/posts/reactive-programming-in-financial-markets/

### Used Package Manager

npm

### System Info

```shell
Not relevant
```

**Comment by meteorlxy** [comment]
Yep, [markdown-it-anchor](https://github.com/valeriangalliat/markdown-it-anchor#permalinks) has provided multiple solutions for that since v8.0.

We didn't update to use other methods to get better a11y yet. Contribution welcome
